.github/workflows/caching.yml

```diff
@@ -165,25 +162,24 @@ jobs:
 
       - if: steps.compiled-deps.outputs.cache-hit != 'true'
         run: |
-          cabal update
+          cabal v2-update
 
       - if: steps.compiled-deps.outputs.cache-hit != 'true'
         name: Download all sources
         run: |
-          cabal $cabalBuild --only-download
+          cabal $cabalBuild --only-download || cabal v2-build all --enable-tests --only-download || cabal v2-build all --enable-benchmarks --only-download || cabal v2-build all || cabal v2-build --enable-tests --enable-benchmarks || cabal v2-build --enable-tests
 
       # repeating builds to workaround segfaults in windows and ghc-8.8.4
       # This build agenda in not to have successful code,
       # but to cache what can be cached, so step is fault tolerant & would always succseed.
       # 2021-12-11: NOTE: Building all targets, since
       # current Cabal does not allow `all --enable-tests --enable-benchmarks --only-dependencies`
+      #
+      # Additional commands are: to caching anything possible when `tests` deeps do not match with `benchmarks`
+      # or when manually run inside inconsistent dep conf PRs to return a positive error code eventually.
+      # Reminder: the `||` work analog to `<|>`, running comments from the left.
       - if: steps.compiled-deps.outputs.cache-hit != 'true'
         name: Build all targets; try 3 times
         continue-on-error: true
         run: |
-          cabal $cabalBuild || cabal $cabalBuild || cabal $cabalBuild
-
-      # Despite the `continue-on-error: true` directive - CI does not ignore the return code of the last step
-      - name: Workaround to CI platform
-        run: |
-          true
+          cabal $cabalBuild || cabal $cabalBuild || cabal $cabalBuild || cabal v2-build all --enable-tests --keep-going || cabal v2-build all --enable-benchmarks --keep-going || cabal v2-build all --keep-going || cabal v2-build --enable-tests --enable-benchmarks --keep-going || cabal v2-build --enable-tests --keep-going || cabal v2-build --keep-going
```

Comment:
Not sure about this. I think we should optimize workflows for the "happy" path, the stable master branch and pr's not tweaking too much the project dependencies. 
`ghc-9.2.1` branch is somewhat a extraordinary one and as pepe commented there, optimize caching for it is not the main goal, but make it work. Authors of commits are maintainers testing changes locally or in their forks and fast ci is desirable but no essential imo.

Reply:
Well, yes this is also true.

This would've been a nice thing to have simply because, since we have manual runs inside PRs - this would be logical to have to cache through difficulties in the PR.

While `||` also ensures that after success commands on the right do not run, CI in `master` would terminate on first 4-5 of them - which would create a good cache. & that is why (because it is free to be added and skipped over) I was free to make that long line of commands "just in case" it runs inside PRs.

But at this point I also have no interest, that is a minor thing, this PR has what to attend to: *(updated the head post)*

Reply:
But whould note, this was also the way to follow the provided by `test` & `bench` guarantees, while not requiring maintenance of https://github.com/haskell/haskell-language-server/pull/2536#issuecomment-1001551417